components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -55,18 +72,57 @@ func main() {
 		provisionerClient = provisioner.NewProvisionerClient(cfg.Provisioning.URL, true)
 	}
 
+	secrets, err := newSecretsInterface(cfg.Director.Namespace)
+	fatalOnError(err)
+	oauthClient := oauth.NewOauthClient(newHTTPClient(false), secrets, cfg.Director.OauthCredentialsSecretName)
+	fatalOnError(oauthClient.WaitForCredentials())
+	tkn, err := oauthClient.GetAuthorizationToken()
+	fatalOnError(err)
+	fmt.Println("DUPA:", tkn.AccessToken)
+	director.NewDirectorClient(oauthClient)
+
 	db, err := storage.New(cfg.Database.ConnectionURL())
 	fatalOnError(err)
 
-	broker, err := broker.NewBroker(provisionerClient, cfg.Provisioning, db.Instances())
+	kymaEnvBroker, err := broker.NewBroker(provisionerClient, cfg.Provisioning, db.Instances())
 	fatalOnError(err)
 
-	brokerAPI := brokerapi.New(broker, logger, brokerCredentials)
+	brokerAPI := brokerapi.New(kymaEnvBroker, logger, brokerCredentials)
 	r := handlers.LoggingHandler(os.Stdout, brokerAPI)
 
 	fatalOnError(http.ListenAndServe(cfg.Host+":"+cfg.Port, r))
 }
 
+func newHTTPClient(skipCertVeryfication bool) *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: skipCertVeryfication},
+		},
+		Timeout: 30 * time.Second,
+	}
+}
+
+func newSecretsInterface(namespace string) (v1.SecretInterface, error) {
+	k8sConfig, err := restclient.InClusterConfig()
+	if err != nil {
+		logrus.Warnf("Failed to read in cluster config: %s", err.Error())
+		logrus.Info("Trying to initialize with local config")
+		home := homedir.HomeDir()
+		k8sConfPath := filepath.Join(home, ".kube", "config")
+		k8sConfig, err = clientcmd.BuildConfigFromFlags("", k8sConfPath)
+		if err != nil {
+			return nil, errors.Errorf("failed to read k8s in-cluster configuration, %s", err.Error())
+		}
+	}
+
+	coreClientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		return nil, errors.Errorf("failed to create k8s core client, %s", err.Error())
+	}
+
```

Comment:
Is it not easier to use:
```
cfg, err := config.GetConfig()
k8sCli, err := client.New(cfg, client.Options{})
```
from:
"sigs.k8s.io/controller-runtime/pkg/client"
"sigs.k8s.io/controller-runtime/pkg/client/config"
libraries?